apps/antalmanac/src/components/RightPane/CoursePane/SearchForm/CourseNumberSearchBar.tsx

```diff
@@ -14,24 +14,40 @@ class CourseNumberSearchBar extends PureComponent<Record<string, never>, CourseN
     }
 
     getCourseNumber() {
-      return RightPaneStore.getUrlCourseNumValue().trim() 
-        ? this.updateCourseNumAndGetFormData() 
-        : RightPaneStore.getFormData().courseNumber;
+        return RightPaneStore.getUrlCourseNumValue().trim()
+            ? this.updateCourseNumAndGetFormData()
+            : RightPaneStore.getFormData().courseNumber;
     }
 
     state = {
         courseNumber: this.getCourseNumber(),
     };
 
-    handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-        this.setState({ courseNumber: event.target.value });
-        RightPaneStore.updateFormValue('courseNumber', event.target.value);
+    handleChange = (name: string) => (event: ChangeEvent<HTMLInputElement>) => {
         const url = new URL(window.location.href);
         const urlParam = new URLSearchParams(url.search);
         urlParam.delete('courseNumber');
-        if (event.target.value) {
-            urlParam.append('courseNumber', event.target.value);
+
+        if (name === 'upper') {
+            if (event.target.checked) {
+                this.setState({ courseNumber: '100-199' });
```

Comment:
Good changes, this function is more readable now. Now that I can see more clearly what it's doing, I think a better design would be to leave `handleChange` as it was before this PR, and make another function `handleUpperDivChange` that has some simple logic like 
```
if(event.target.checked) {
    handleChange({target: {value: '100-199'}})
else {
    handleChange({target:{}})
}
```
This makes it more modular, and you could also rename `handleChange` to be something like `handleNumberChange` to be more specific since it only has one job.
This would also let you try to make the type on `event` more strict on both of the functions since they're different types of input elements (TextField vs Switch instead of a generic HTMLInputElement). I'm not sure if more fine-grained types exist but if they do we should use them.

Let me know what you think of these suggestions. I get that this is kinda overkill for a small PR but the idea is to practice code quality so you can write code that's more readable for bigger changes.